proto/ocpaths.proto

```diff
@@ -62,6 +75,24 @@ message OCPath {
   // GNMIRpc describes expected (or supported) behavior for a particular
   // Openconfig path.
   GNMIRpc gnmi_rpc = 5;
+
+  // the date/time when this path became supported. not present if 
+  // unknown.
+  optional Release origination = 6;
+
+  // the date/time when this path became deprecated. not present for active
+  // paths. origination must be defined (not enforced). not mutually 
+  // exclusive with origination and removal.
+  optional Release deprecation = 7;
+
+  // the date/time when this path was removed from support. not present
+  // for active paths or if unknown. origination must be defined (not enforced).
+  // not mutually exclusive with deprecation.
+  optional Release removal = 8;
+
+  // the YANG model version that first defined this path. this is typically
```

Comment:
is there any value in tracking this for each vendor/sw/hw release? looks like vendors will need to maintain a database with such information in order export it..?

or is the expectation that the data will be provided by OC community and not used by vendors?

Reply:
Not the OC community - this is vendor-specific. I'll update the comments to make this clear.
That database would be the NOS images from previous releases. External logic would walk those
NOS images and calculate any origination/deprecation/removal dates to populate the current NOS
image. As a vendor, it is fairly common for our customers to ask for this information. But it is marked
as optional (I saw your later comment on this - I'll comment there) as a reminder that a vendor can
release a NOS image without this information.

Reply:
>Not the OC community - this is vendor-specific.
>  // the YANG model version that first defined this path.

In that case perhaps I don't understand how the "yang model version" is defined. We, as a vendor, don't have any additional versioning of OC models. Or perhaps you're talking about your PR in general but not about the specific field?
That question is related to the field only (unfortunately I quoted the first line of the comment which didn't produce a nice preview). As for the overall PR reasoning, I left a comment above.

Reply:
that yangModelVersion typically identifies the OpenConfig consortium model that defines this leaf.

Reply:
>that yangModelVersion typically identifies the OpenConfig consortium model that defines this leaf.

That brings me back to the original question: what is the value of tracking it per vendor/platform? :-)

Reply:
tracking this was a specific requirement handed to us by G

Reply:
@dplore would you be able to provide context?

Reply:
[NOSImageProfile](https://github.com/openconfig/featureprofiles/blob/main/proto/nosimage.proto) should be used to describe the contents of an operating system's support.

A series of NOSImageProfile messages (textproto files) can be used to learn when a path is introduced and removed.   

Any fields related to the  NOS support should be present in the NOSImageProfile proto.

Reply:
If this field is intended to represent the OC yang model version, then when used inside of a NOSImageProfile, this field can assert the OC yang model version the NOS supports.  I am not sure it has to be the 'earliest'.  It can merely be whatever version the NOS was built to support.